src/JsonSchema/Constraints/BaseConstraint.php

```diff
@@ -54,18 +61,42 @@ public function addError(JsonPointer $path = null, $message, $constraint = '', a
         }
 
         $this->errors[] = $error;
+        $this->errorMask |= $error['context'];
     }
 
     public function addErrors(array $errors)
     {
         if ($errors) {
             $this->errors = array_merge($this->errors, $errors);
+            $errorMask = &$this->errorMask;
+            array_walk($errors, function ($error) use (&$errorMask) {
+                if (isset($error['context'])) {
+                    $errorMask |= $error['context'];
+                }
+            });
         }
     }
 
-    public function getErrors()
+    public function getErrors($errorContext = null)
```

Comment:
Just a quick comment before I head to work; can we have a more meaningful default value? It would make it a little more clear to the reader how this method works. Something like:

```
public function getErrors($errorContext = Validator::ERROR_ALL);
```